src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,112 @@
+import duke.tasks.Task;
+import duke.tasks.TaskType;
+
+/**
+ * Represents a storage that read from and writes to a file.
+ * @version 1.0
+ */
+public class Storage {
+    private final File file;
+
+    /**
+     * Creates a new Storage object with the given file path.
+     *
+     * @param filePath A String representation of the target file path.
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Finds the file through file path reference.
+     * Translates the file into an ArrayList of tasks.
+     *
+     * @return The ArrayList of tasks translated from the file.
+     * @throws DukeException thrown if file not found.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> lst = new ArrayList<>();
+
+        try {
+            Scanner fileReader = new Scanner(file);
+            while (fileReader.hasNextLine()) {
+                String line = fileReader.nextLine();
+                String[] command = line.split(" \\| ");
+                switch (command[0]) {
+                case "T":
+                    Task todo = new Task(TaskType.TODO, command[2]);
+                    if (command[1].equals("1")) {
+                        todo.setDone();
+                    }
+                    lst.add(todo);
+                    break;
+                case "D":
+                    Task deadline = new Task(TaskType.DEADLINE, command[2], LocalDate.parse(command[3]));
+                    if (command[1].equals("1")) {
+                        deadline.setDone();
+                    }
+                    lst.add(deadline);
+                    break;
+                case "E":
+                    Task event = new Task(TaskType.EVENT, command[2], LocalDate.parse(command[3]));
+                    if (command[1].equals("1")) {
+                        event.setDone();
+                    }
+                    lst.add(event);
+                    break;
+                default:
+                }
+            }
+            fileReader.close();
+        } catch (IOException e) {
+            throw new DukeException("\tFile not Found.");
+        } catch (Exception e) {
+            throw new DukeException("\tError loading history. " + e.getMessage());
+        }
+
+        return lst;
+    }
+
+    /**
+     * Creates the directory if the targeted file path does not exist.
+     * Translates the ArrayList of tasks into a String representation.
+     * Writes the String representation of the task list into the targeted file.
+     *
+     * @param lst the ArrayList of tasks from some TaskList.
+     * @throws DukeException thrown if error during writing to file.
+     */
+    public void write(ArrayList<Task> lst) throws DukeException {
+        try {
+            file.getParentFile().mkdir();
+            FileWriter fileWriter = new FileWriter(file);
+            String listToString = "";
+            for (Task t : lst) {
```

Comment:
Perhaps to improve code readability, there could be a separate method which composes the string to be written to the file?